### DISCUSSION TITLE
Is QZ Fitness able to register a workout and import it into Apple Health/Fitness

### DISCUSSION THREAD

**Discussion by detro** [cagnulein/qdomyos-zwift#2015] [Q&A]
So, my setup so it helps understand why I'm asking this:

1. Echelon EX3 (just bought)
2. I don't want to pay for Echelon £30/pm subscription: too much
3. I don't mind subscribing to Apple Fitness+ (£9.99/pm, more reasonable)
4. Garmin Forerunner 245 (I can use it to broadcast my heartrate)
5. **I purchased QZ Fitness today**

So, I can start QZ and connect it to the EX3 no problem (excellent pairing speed).

How do ensure that, once I'm done with a workout, it goes into my Apple Fitness+/Health data?

Alternatively, could I go the Garmin route and use Garmin Connect as gateway?

Maybe I should have already worked all this out, but maybe my 42yo brain is getting mushy and stupid.

Thanks for any help/advice

**Comment by cagnulein** [comment]
hi unfortunately you need an apple watch for writing on apple health.
for the garmin, you can push the fit file from qz (that you can receive at
the end of the ride by email) to the garmin connect (so you can also sync
to apple health probably)
let me know if you have any other questions

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 19 gen 2024 alle 21:07 Ivan De Marino <
***@***.***> ha scritto:

**Reply by detro** [reply]
Ah, so for now no luck to push directly to Apple Health. I see.

Did I notice it right that you have an ongoing branch trying to bridge this gap?

Anyway, meantime I will give the QZ->email FIT->Garmin upload->Apple Health route a go.

Thank you

**Comment by cagnulein** [comment]
yes actually that worked, if you want i can send you the experimental
version. send me your email to roberto.viola83 at gmail.com



Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 19 gen 2024 alle 22:28 Ivan De Marino <
***@***.***> ha scritto:

**Reply by cagnulein** [reply]
i have to build it again in the next days

**Comment by detro** [comment]
So, I tried the "get the Fit via email, upload it to Garmin and sync with Apple": it works.

Even better if we can go directly from QZ to Apple, of course.

I'm sending you an email.

**Reply by cagnulein** [reply]
perfect!

**Comment by dclive** [comment]
Is there documentation that goes over all of this in a little more detail?  

I have an Apple Watch, am buying the Echelon EX5, and just got the QZ app; I plan to use MyWhoosh and Apple Fitness +.  

What should my expectations be on what my watch does with each / how will that work?  Is there a (current / updated) version of setup guides for popular scenarios (AppleTV + Apple Watch + Mac to broadcast to AppleTV, with MyWhoosh, or AppleTV + Apple Watch + Apple Fitness +) and required setup steps?

**Reply by cagnulein** [reply]
hi @dclive simply use QZ on your phone and watch too and QZ will write directly the workout on apple health. Then connect QZ for mywhoosh on another device. For apple firntess, just use apple fitness on your tv.
